### ISSUE TITLE
[Pattern] BatchNorm training=True -> False

### ISSUE THREAD

**Issue by justinchuby** [microsoft/onnxscript#1791]
torch.export sometimes creates BatchNorm with training=True even though the model is an inference model. Create a rewrite rule to set training to False.

Related: https://github.com/microsoft/onnxscript/issues/1338

**Comment by titaiwangms** [comment]
hmm this is hard to solve because we wouldn't know the training mode is set intentionally or it's forced to be changed to.

**Comment by titaiwangms** [comment]
And it seems that torch.export.export not only changes it to training mode, but also changes `running_var` and `running_mean`.  If we simply change the node:     
```
sess = C.InferenceSession(session_options, self._model_path, True, self._read_config_from_model)
onnxruntime.capi.onnxruntime_pybind11_state.Fail: [ONNXRuntimeError] : 1 : FAIL : Load model from dump_bash_bench/Generator-torch-onnx-cuda-torch.float16/model.onnx failed:Node (node__aten_native_batch_norm_inference_onnx_95) Op (_aten_native_batch_norm_inference_onnx) [ShapeInferenceError] Inference error(s): (op_type:BatchNormalization, node name: n0): [ShapeInferenceError] Input 3 expected to have rank 1 but has rank 0 in node BatchNormalization (n0).
```

**Comment by titaiwangms** [comment]
Found this is training related issue. It's possible that aot_autograd generates this purposely.

**Comment by titaiwangms** [comment]
Eventually, users would see

```
  File "/root/onnxruntime/build/Linux/RelWithDebInfo/onnxruntime/capi/onnxruntime_inference_collection.py", line 491, in _create_inference_session
    sess.initialize_session(providers, provider_options, disabled_optimizers)
onnxruntime.capi.onnxruntime_pybind11_state.RuntimeException: [ONNXRuntimeError] : 6 : RUNTIME_EXCEPTION : Exception during initialization: /root/onnxruntime/onnxruntime/core/providers/cuda/nn/batch_norm.h:44 onnxruntime::cuda::BatchNorm<T, NCHW>::BatchNorm(const onnxruntime::OpKernelInfo&) [with T = onnxruntime::MLFloat16; bool NCHW = false] !(is_training_mode_ && opset >= 14) was false. Training mode does not support BN opset 14 (or higher) yet.
```
